### PR TITLE
API Plugin Support

### PR DESCRIPTION
**Pull request by TheFynx**
# Final MVP Item
- API Plugin
  - Can now add API plugins natively or be bot controlled

- Build From Dockerfile
  - Can now build container and api plugins from a Dockerfile (#49)

- Container Management Options
  - Can choose to have container type plugins not managed by the bot as longer as they are on the same host
  - Can Choose to have managed api plugins

- Cleanup
  - During shutdown the bot now cleans up any of it's managed items (containers/scripts)

### CODE REVIEW

File: lib/cleanup/helpers/language.rb [hunk 1]
```diff
@@ -0,0 +1,23 @@
+# frozen_string_literal: true
+
+# Docker Helpers for Plugin Class
+# Its main functions are to:
+#  1. Set Items to be binded/mounted to Plugin Container
+#     - Include script if script type
+#     - Include plugin config if configured with path to mount/bind
+#  2. Set Script Language based on plugin config
+#     - Language determines which image is pulled for script exec
+class Cleanup
```
Comment (aaronblythe) [inline lib/cleanup/helpers/language.rb:10]: Curious if this should be moved to a reusable location for both Plugins and Cleanup to not repeat?

File: config.ru [hunk 2]
```diff
@@ -1,3 +1,12 @@
 # frozen_string_literal: true
+require_relative 'lib/cleanup/cleanup.rb'
+
+at_exit do
+  cleanup = Cleanup.new
+  cleanup.shutdown
```
Comment (aaronblythe) [inline config.ru:6]: Can any exceptions be thrown from this?  Or can any indication be made on the number of containers still running on the host (hopefully zero).

**Review by aaronblythe — APPROVED** [review]

File: config.ru (hunk 2 above)
Comment (TheFynx) [reply config.ru:6]: Oh it throws exceptions if something goes wrong, took me forever to get it right :-)

File: lib/cleanup/helpers/language.rb (hunk 1 above)
Comment (TheFynx) [reply lib/cleanup/helpers/language.rb:10]: I'll add an issue to look at this next sprint.

File: lib/brain/redis.rb [hunk 3]
```diff
@@ -77,7 +77,7 @@ def set_url
     container_ip = @container_info['NetworkSettings']['IPAddress']
 
     # Determine if running via DIND/Compose Config or if running local
-    compose_bot = false unless ip.rpartition('.')[0] == container_ip.rpartition('.')[0]
+    compose_bot = true if ip.rpartition('.')[0] == container_ip.rpartition('.')[0]
```
Comment (aaronblythe) [inline lib/brain/redis.rb]: I believe the 'true if' is extraneous.

```
2.3.0 :001 > apples = 'apples' == 'apples'
 => true 
2.3.0 :002 > apples = 'apples' == 'bananas'
 => false 
```

**Review by aaronblythe — APPROVED** [review]

**Pull request merged**